item-impl/src/main/scala/com/example/auction/item/impl/ItemRepository.scala

```diff
@@ -50,6 +53,26 @@ private[impl] class ItemRepository(session: CassandraSession)(implicit ec: Execu
     }
   }
 
+  /**
+    * Motivation: https://discuss.lightbend.com/t/how-to-specify-pagination-for-select-query-read-side/870
+    */
+  private def selectItemsByCreatorInStatusWithPaging(creatorId: UUID, status: api.ItemStatus.Status, offset: Int, limit: Int) = {
+
+    implicit val system: ActorSystem = ActorSystem("ItemActorSystem")
+    implicit val materializer: ActorMaterializer = ActorMaterializer()
```

Comment:
Don't create a new `ActorSystem`. If you need one it should be injected in the constructor and it will be provided by `macwire` in `ItemComponents`.

Reply:
`ActorMaterializer` should also be injected.

Reply:
🤦‍♂️ absolutely! Forgot to mention that.

Reply:
Thanks @ignasi35 and @renatocaval for the feedback. I checked `ItemComponents` and Lagom provides a Materializer in `LagomServerComponents` so I reused it defining `implicit def materializer: Materializer` in `ItemComponents`. I removed the lines on `ItemRepository` and added Materializer as implicit.